fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -508,6 +508,19 @@ ACTOR Future<Void> releaseResolvingAfter(ProxyCommitData* self, Future<Void> rel
 	return Void();
 }
 
+static void inspectTransaction(const CommitTransactionRequest& tr, bool* hasMetadataMutations,
```

Comment:
We already iterate through the transactions and its mutations multiple times (and `commitBatch` is a hot code-path).

Could this logic be incorporated into `ResolutionRequestBuilder::addTransaction`?

Reply:
Done